### ISSUE TITLE
Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFile.createUnique]

### ISSUE THREAD

**Issue by gldickens3** [mlazdans/qnote#53]
I've been using Qnote for years but I recently ran into a bug running Tbird 115.12.2 on Ubuntu Linux. Every time that I try to create a new Qnote then Thunderbird will note save the note and I get this error:

Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFile.createUnique]

Any suggestions for  me or is this just a bug that the developer needs to fix?

Thanks,

Gordon

**Comment by mlazdans** [comment]
Please check if thunderbird process has permissions to write into note folder

**Comment by gldickens3** [comment]
The thunderbird process is running as my user name, gordon, and all of the files and directories in my home directory, including the .thunderbird directory, are owned by user gordon and group gordon..  So, my thunderbird process definitely has permissions to write into the note folder, assuming that the note folder is in my home directory.  BTW, where is the notes folder stored?

**Comment by mlazdans** [comment]
It depends.  Please check settings:
![image](https://github.com/mlazdans/qnote/assets/48044624/6d144250-e496-4ea8-ac19-c61a7067f158)

**Comment by gldickens3** [comment]
Thanks mlazdans!  It was not a permissions problem but I fixed everthing as follows.   Ubuntu recently changed thunderbird to be a snap package and the installation location of thunderbird changed from ~/.thunderbird to ~/snap/thunderbird/common/.thunderbird, however, my QNote extension was still pointed to the old location, ~/.thunderbird, so I changed it to the new location, ~/snap/thunderbird/common/.thunderbird after which it started working properly and it now will save my QNotes successfully and without any error messages.  Problem  solved!
